pkg/rtc/participant.go

```diff
@@ -818,35 +817,45 @@ func (p *ParticipantImpl) GetAudioLevel() (level float64, active bool) {
 }
 
 func (p *ParticipantImpl) GetConnectionQuality() *livekit.ConnectionQualityInfo {
-	publisherScores := p.getPublisherConnectionQuality()
+	numTracks := 0
+	minQuality := livekit.ConnectionQuality_EXCELLENT
+	minScore := float32(0.0)
 
-	numTracks := len(publisherScores)
-	totalScore := float32(0.0)
-	for _, score := range publisherScores {
-		totalScore += score
+	for _, pt := range p.GetPublishedTracks() {
+		numTracks++
+
+		score, quality := pt.(types.LocalMediaTrack).GetConnectionScoreAndQuality()
+		if quality < minQuality {
+			// WARNING NOTE: comparing protobuf enums directly
+			minQuality = quality
+			minScore = score
+		} else if quality == minQuality && score < minScore {
+			minScore = score
+		}
 	}
 
 	subscribedTracks := p.SubscriptionManager.GetSubscribedTracks()
-	subscriberScores := make(map[livekit.TrackID]float32, len(subscribedTracks))
 	for _, subTrack := range subscribedTracks {
-		if subTrack.IsMuted() || subTrack.MediaTrack().IsMuted() {
-			continue
-		}
```

Comment:
Removing the muted check from here. The quality scorer is notified of mute transitions and reports quality accordingly (i. e. when muted `EXCELLENT` is reported). So, leaving it completely to quality scorer to report based on its current state.